src/aggregation/agg_limits.rs

```diff
@@ -91,8 +93,6 @@ impl AggregationLimits {
         Ok(())
     }
     pub(crate) fn add_memory_consumed(&self, num_bytes: u64) {
```

Comment:
Can we merge `add_memory_consumed` and `validate_memory_consumption` and  internalize the computation of the delta instead of having the logic everywhere?

```fn set_memory_consumed(&mut self, num_bytes: u64) -> crate::Result<()>

Reply:
I thought about internalizing the delta logic, as I don't like the replication either, but it can't be part of the guard, since the cardinality is different

The idea behind splitting `add_memory_consumed` and `validate_memory_consumption` is to have multiple `add_memory_consumed` calls and a more expensive `validate_memory_consumption` call, but it's never used like this currently.

Reply:
> I thought about internalizing the delta logic, as I don't like the replication either, but it can't be part of the guard, since the cardinality is different

Ah yes indeed.

Reply:
> The idea behind splitting add_memory_consumed and validate_memory_consumption is to have multiple add_memory_consumed calls and a more expensive validate_memory_consumption call, but it's never used like this currently.

If it is still not used can we shoot for factorized API?
`.add_memory_consumed(add_num_bytes: u64) -> Result<(), AggregationError>`

Reply:
yep, updated